src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -194,23 +193,28 @@ internal override bool IsCallerMemberName
             }
         }
 
-        private ConstantValue DefaultSyntaxValue
+        private ConstantValue GetDefaultSyntaxValue(DiagnosticBag diagnosticsOpt = null)
```

Comment:
Do you think this is encompassed by https://github.com/dotnet/roslyn/issues/16652?

That would remove the flakiness with adding to declaration diagnostics.